core/src/main/java/bio/terra/pearl/core/service/workflow/TaskConfigDispatcher.java

```diff
@@ -0,0 +1,347 @@
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * builds a task for the given survey -- does NOT evaluate the rule or check duplicates
+     */
+    public ParticipantTask buildTask(Enrollee enrollee,
+                                     PortalParticipantUser portalParticipantUser,
+                                     T taskConfig) {
+        TaskType taskType = getTaskType(taskConfig);
+        ParticipantTask task = ParticipantTask.builder()
+                .enrolleeId(enrollee.getId())
+                .portalParticipantUserId(portalParticipantUser.getId())
+                .studyEnvironmentId(enrollee.getStudyEnvironmentId())
+                .blocksHub(taskConfig.isRequired())
+                .taskOrder(taskConfig.getTaskOrder())
+                .targetStableId(taskConfig.getStableId())
+                .targetAssignedVersion(taskConfig.getVersion())
+                .taskType(taskType)
+                .targetName(taskConfig.getName())
+                .status(TaskStatus.NEW)
+                .build();
+        return task;
+    }
+
+    public boolean isEligible(T taskDispatchConfig, EnrolleeContext enrolleeContext) {
+        /**
+         * eligible if the enrollee is a subject, the task is not restricted by time, and the enrollee meets the rule
+         * note that this does not include a duplicate task check -- that is done elsewhere
+         */
+        // TODO JN-977: this logic will need to change because we will need to support surveys for proxies
+        return enrolleeContext.getEnrollee().isSubject() &&
+                (taskDispatchConfig.getDaysAfterEligible() == null ||
+                        enrolleeContext.getEnrollee().getCreatedAt().plus(taskDispatchConfig.getDaysAfterEligible(), ChronoUnit.DAYS).isBefore(Instant.now())) &&
+                enrolleeSearchExpressionParser
+                        .parseRule(taskDispatchConfig.getEligibilityRule())
+                        .evaluate(new EnrolleeSearchContext(enrolleeContext.getEnrollee(), enrolleeContext.getProfile()));
+    }
+
+
+    protected List<Enrollee> findMatchingEnrollees(ParticipantTaskAssignDto assignDto,
+                                                   UUID studyEnvironmentId) {
+        if (assignDto.assignAllUnassigned()
+                && (Objects.isNull(assignDto.enrolleeIds()) || assignDto.enrolleeIds().isEmpty())) {
+            return enrolleeService.findUnassignedToTask(studyEnvironmentId,
+                    assignDto.targetStableId(), null);
+        } else {
+            return enrolleeService.findAll(assignDto.enrolleeIds());
+        }
+    }
+
+    /**
+     * To avoid accidentally assigning the same tasks or outreach activity to a participant multiple times,
+     * confirm that if the stableId matches an existing task, the existing task must be complete and the
+     * configured survey must allow recurrence.
+     */
+    public boolean isDuplicateTask(T taskDispatchConfig, ParticipantTask task,
```

Comment:
don't forget this

Reply:
Oops, sorry, I forgot to leave a comment. This is actually being used in a few `SurveyTaskDispatcher` tests, so I thought it would be easier to just keep it public rather than delete the tests.